[BUG] mlx::core::version() is not exported from the mlx static library

**Describe the bug**

* Callers of the `mlx` static library are unable to call `mlx::core::version()` as the symbol is not defined by `mlx`.

**To Reproduce**

Build the `mlx` library and use `nm` to view the symbols
```bash
# Running against main (commit 8576e6fe3606bf5b805162fd5f4a7803a9a0d349) 
# cd to root of mlx
mkdir -p build && cd build
cmake .. -DCMAKE_BUILD_TYPE=Release
make -j
nm -C libmlx.a | grep mlx::core::version
```

Output:
```
                 U mlx::core::version()
```

---

Attempt to use the `mlx` library through `mlx-c` `examples/example.c`:
```bash
# Running against mlx-c main (commit 9ebe155864eab06d94ba18e01f9cb2666b2975a7)
# cd to root of mlx-c
cc -I . -L <mlx/mlxc path> -lmlxc -lmlx -framework Metal -framework Foundation -framework CoreFoundation -framework QuartzCore -framework Accelerate -lobjc -lc++ -mmacosx-version-min=15.4 examples/example.c
```

Output:
```
Undefined symbols for architecture arm64:
  "mlx::core::version()", referenced from:
      _mlx_version in libmlxc.a[25](version.cpp.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Expected behavior**

* `mlx::core::version()` is exported by `mlx` such that consumers can call `mlx::core::version()`

**Desktop (please complete the following information):**
 - OS Version: MacOS 15.4
 - Version: `8576e6fe3606bf5b805162fd5f4a7803a9a0d349` (main)

**Additional context**

* I'm not a CMake expert, but my understanding is that the changes in https://github.com/ml-explore/mlx/pull/1966 caused this regression.
    * `target_link_libraries(mlx PRIVATE $<BUILD_INTERFACE:mlx_version>)` uses `mlx_version` privately such that the symbols from `mlx_version` are available within `mlx` but are not exported to consumers of `mlx`.

I believe I have a minimal reproducer that demonstrates the issue. Linking is indeed part of the issue, but only `mlx::core::version()` is not present in the static library due to the quirk from https://github.com/ml-explore/mlx/pull/1966.

## Reproduction Steps

* Follow the build instructions for MLX and MLX-C [as mentioned in the README.md](https://github.com/ml-explore/mlx-c?tab=readme-ov-file#install)
  * i.e `mkdir build && cd build; cmake .. -DCMAKE_BUILD_TYPE=Release; make -j`
* Save the following exemplars:

**version_example.c**
```
#include <stdio.h>
#include "mlx/c/mlx.h"

int main() {
  mlx_string version = mlx_string_new();
  mlx_version(&version);
  printf("MLX version: %s\n", mlx_string_data(version));
  mlx_string_free(version);
  return 0;
}
```

**gpu_info_example.c**
```
#include <stdio.h>
#include "mlx/c/mlx.h"

int main() {
    printf("==================================================\n");
    printf("GPU info:\n");
    mlx_metal_device_info_t info = mlx_metal_device_info();
    printf("architecture: %s\n", info.architecture);
    printf("max_buffer_length: %ld\n", info.max_buffer_length);
    printf(
        "max_recommended_working_set_size: %ld\n",
        info.max_recommended_working_set_size);
    printf("memory_size: %ld\n", info.memory_size);

    printf("==================================================\n");
  return 0;
}
```

---

* Attempt to compile `gpu_info_example`: 

```
clang -I /path/to/mlx-c -L /path/to/mlx-c/build -L /path/to/mlx/build -lmlxc -lmlx -framework  Metal -framework Foundation -framework Accelerate -lc++ gpu_info_example.c -o gpu_info_example
```

* `gpu_info_example` builds successfully

---

* Attempt to compile `version_example`:

```
clang -I /path/to/mlx-c -L /path/to/mlx-c/build -L /path/to/mlx/build -lmlxc -lmlx -framework  Metal -framework Foundation -framework Accelerate -lc++ version_example.c -o version_example
```

*  `version_example` does not build successfully:

```
Undefined symbols for architecture arm64:
  "mlx::core::version()", referenced from:
      _mlx_version in libmlxc.a[25](version.cpp.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/mlx/CMakeLists.txt
+++ b/mlx/CMakeLists.txt
@@ -21,7 +21,7 @@ target_sources(
           ${CMAKE_CURRENT_SOURCE_DIR}/backend/metal/metal.h)
 
 # Define MLX_VERSION only in the version.cpp file.
-add_library(mlx_version STATIC ${CMAKE_CURRENT_SOURCE_DIR}/version.cpp)
+add_library(mlx_version OBJECT ${CMAKE_CURRENT_SOURCE_DIR}/version.cpp)
 target_compile_definitions(mlx_version PRIVATE MLX_VERSION="${MLX_VERSION}")
 target_link_libraries(mlx PRIVATE $<BUILD_INTERFACE:mlx_version>)
 

